File: jobs/2026-05-07__08-46-23/reminder_workflow_py__X33eFJM/artifacts/code/start.py
import os
import asyncio
import uuid
from temporalio.client import Client
from workflows import ReminderWorkflow

async def main():
    address = os.getenv("TEMPORAL_ADDRESS")
    namespace = os.getenv("TEMPORAL_NAMESPACE")
    api_key = os.getenv("TEMPORAL_API_KEY")

    if not address or not namespace or not api_key:
        raise ValueError("TEMPORAL_ADDRESS, TEMPORAL_NAMESPACE, and TEMPORAL_API_KEY must be set")

    client = await Client.connect(
        address,
        namespace=namespace,
        api_key=api_key,
        tls=True,
    )

    result = await client.execute_workflow(
        ReminderWorkflow.run,
        args=[2, "Time is up!"],
        id=f"reminder-workflow-{uuid.uuid4()}",
        task_queue="reminder-task-queue",
    )

    print(f"RESULT: {result}")

if __name__ == "__main__":
    asyncio.run(main())

File: jobs/2026-05-27__01-32-13/workflow_update_handler_ts__U85fyem/artifacts/user/myproject/src/start.ts
import { Connection, Client } from '@temporalio/client';
import { NativeConnection, Worker } from '@temporalio/worker';
import path from 'path';
import { BankBalanceWorkflow } from './workflows';

const taskQueue = 'update-handler-ts';

function getRequiredEnv(name: string): string {
  const value = process.env[name];
  if (!value) {
    throw new Error(`Missing required environment variable: ${name}`);
  }
  return value;
}

async function runClient(): Promise<void> {
  const address = getRequiredEnv('TEMPORAL_ADDRESS');
  const apiKey = getRequiredEnv('TEMPORAL_API_KEY');
  const namespace = getRequiredEnv('TEMPORAL_NAMESPACE');
  const runId = getRequiredEnv('ZEALT_RUN_ID');

  const connection = await Connection.connect({
    address,
    tls: true,
    apiKey,
  });

  const client = new Client({ connection, namespace });
  const workflowId = `update-wf-${runId}`;

  const handle = await client.workflow.start(BankBalanceWorkflow, {
    taskQueue,
    workflowId,
  });

  const amounts = [100, 50, 25];
  for (const amount of amounts) {
    const updatedBalance = await handle.executeUpdate<number>('deposit', {
      args: [amount],
    });
    console.log(`Updated balance: ${updatedBalance}`);
  }

  await handle.signal('finish');

  const finalBalance = await handle.result();
  console.log(`Final balance: ${finalBalance}`);
}

async function main(): Promise<void> {
  const address = getRequiredEnv('TEMPORAL_ADDRESS');
  const apiKey = getRequiredEnv('TEMPORAL_API_KEY');
  const namespace = getRequiredEnv('TEMPORAL_NAMESPACE');

  const workerConnection = await NativeConnection.connect({
    address,
    tls: true,
    apiKey,
  });

  const workflowsPath = require.resolve(path.join(__dirname, 'workflows'));

  const worker = await Worker.create({
    connection: workerConnection,
    namespace,
    taskQueue,
    workflowsPath,
  });

  const workerRun = worker.run();

  try {
    await runClient();
  } finally {
    worker.shutdown();
    await workerRun;
  }
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
